app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -940,6 +943,28 @@ class PluginApi {
     extraConnectorClass(`${outletName}/${connectorName}`, klass);
   }
 
+  /**
+   * Register a component to be rendered in a particular outlet.
+   *
+   * For example, if the outlet is `user-profile-primary`, you could register
+   * a component like
+   *
+   * ```javascript
+   * import MyComponent from "discourse/plugins/my-plugin/components/my-component";
+   * api.renderIntoOutlet('user-profile-primary', MyComponent);
+   * ```
+   *
+   * Alternatively, a component could be defined inline using gjs:
+   *
+   * ```javascript
+   * api.renderIntoOutlet('user-profile-primary', <template>Hello world</template>);
+   * ```
+   *
+   */
```

Comment:
Small request, I think it'll be helpful to start documenting the params in the JSDoc format as we add new APIs now. We've been [doing it for the sidebar APIs](https://github.com/discourse/discourse/blob/372eaf49f1327b55edd692e725b0fecb07dd687e/app/assets/javascripts/discourse/app/lib/plugin-api.js#L1938-L1949) and I think the documenting the params help to make the API easier to use for developers.